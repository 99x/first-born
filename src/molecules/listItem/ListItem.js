import React, { Component } from "react";
import { View, StyleSheet, TouchableOpacity, Image, Platform, Dimensions } from "react-native";
import PropTypes from "prop-types";

import { Text } from "../../atoms/Text";
import { Icon } from "../../atoms/Icon";
import { commonColors } from "react-native-atom-elements/src/utils/color";

const { width } = Dimensions.get("window");

export class ListItem extends Component {

    constructor(props) {
        super(props);
        this.state = {
            expanded: false
        };
    }

    render() {
        const { title, description, image, onPress, block, backgroundColor, secondary, ...otherProps } = this.props;
        const { expanded } = this.state;

        const children = React.Children.map(this.props.children, child => child ?
            React.cloneElement(child, { ...child.props, secondary: true }) : null);

        const cardBackColor = secondary ? commonColors.secondaryBackground : backgroundColor;

        const cardStyle = block ? [styles.containerBlock, { backgroundColor: cardBackColor }] : [styles.container, { backgroundColor: cardBackColor }];

        const onListItemTap = children ? this.expandListItem : onPress ? onPress : undefined;

        return (
            <View style={{ width: "100%" }}>
                <TouchableOpacity style={cardStyle} disabled={!onListItemTap} onPress={onListItemTap} {...otherProps}>
                    {image &&
                        <View style={styles.imageContainer}>
                            <Image source={image} style={styles.image} />
                        </View>
                    }
                    <View style={styles.textContainer}>
                        <Text size="h6" color={commonColors.darkGrey} bold >{title}</Text>
                        {description && <Text size="sub_heading" color={commonColors.lightGrey}>{description}</Text>}
                    </View>
                    <View style={styles.imageContainer}>
                        {children &&
                            <Icon name={expanded ? Platform.OS === "ios" ? "arrow-up" : "arrow-dropup" : Platform.OS === "ios" ? "arrow-down" : "arrow-dropdown"} size={20} color={commonColors.inputGrey} />
                        }
                    </View>
                </TouchableOpacity>
                {expanded && children}
            </View>
        )
    }

    expandListItem = () => {
        this.setState({ expanded: !this.state.expanded });
    }

}

ListItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    image: PropTypes.any,
    block: PropTypes.bool,
    backgroundColor: PropTypes.string,
    secondary: PropTypes.bool,
    ...TouchableOpacity.propTypes
}

ListItem.defaultProps = {
    backgroundColor: commonColors.white,
    secondary: false
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "row",
        ...Platform.select({
            android: {
                borderWidth: 0.5,
                borderColor: commonColors.lightGrey,
            },
            ios: {
                borderTopWidth: 0.5,
                borderBottomWidth: 0.5,
                borderTopColor: commonColors.lightGrey,
                borderBottomColor: commonColors.lightGrey,
            }
        }),

        paddingHorizontal: 5
    },
    containerBlock: {
        width,
        flex: 1,
        flexDirection: "row",
        borderTopWidth: 0.5,
        borderBottomWidth: 0.5,
        borderTopColor: commonColors.lightGrey,
        borderBottomColor: commonColors.lightGrey,
        paddingHorizontal: 15
    },
    textContainer: {
        flex: 3,
        flexDirection: 'column',
        padding: 15,
        alignItems: "flex-start"
    },
    imageContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    image: {
        height: 50,
        width: 50,
        borderRadius: 50 / 2
    }
})