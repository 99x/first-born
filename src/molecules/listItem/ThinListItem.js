import React, { Component } from "react";
import {
    View,
    StyleSheet,
    TouchableOpacity,
    Image,
    Platform
} from "react-native";
import PropTypes from "prop-types";

import { Text } from "../../atoms/Text";
import { Icon } from "../../atoms/Icon";
import { commonColors } from "../../utils/color";
import { deviceVariables } from "../../variables/deviceVariables";

export class ThinListItem extends Component {
    render() {
        const {
            title,
            description,
            image,
            icon,
            backgroundColor,
            arrow,
            ...otherProps
        } = this.props;

        const listItemStyle = [styles.container, { backgroundColor }];

        return (
            <TouchableOpacity style={listItemStyle} {...otherProps}>
                {image && (
                    <View style={styles.imageContainer}>
                        <Image style={styles.image} {...image} />
                    </View>
                )}
                {icon && (
                    <View style={styles.imageContainer}>
                        <Icon size={20} {...icon} />
                    </View>
                )}
                <View style={styles.textContainer}>
                    <Text color={commonColors.darkGrey} bold>
                        {title}
                    </Text>
                    {description && (
                        <Text size="footnote" color={commonColors.lightGrey}>
                            {description}
                        </Text>
                    )}
                </View>
                {arrow && (
                    <View style={styles.imageContainer}>
                        <Icon
                            type={
                                Platform.OS === "ios" ? "ionicon" : "material"
                            }
                            name={
                                Platform.OS === "ios"
                                    ? "arrow-forward"
                                    : "keyboard-arrow-right"
                            }
                            size={20}
                            color={commonColors.inputGrey}
                        />
                    </View>
                )}
            </TouchableOpacity>
        );
    }
}

ThinListItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    image: PropTypes.shape({
        ...Image.propTypes
    }),
    icon: PropTypes.shape({
        ...Icon.propTypes
    }),
    backgroundColor: PropTypes.string,
    arrow: PropTypes.bool,
    ...TouchableOpacity.propTypes
};

ThinListItem.defaultProps = {
    backgroundColor: commonColors.white,
    arrow: false
};

const styles = StyleSheet.create({
    container: {
        width: deviceVariables.width,
        flex: 1,
        flexDirection: "row",
        borderTopWidth: 0.5,
        borderBottomWidth: 0.5,
        borderTopColor: commonColors.lightGrey,
        borderBottomColor: commonColors.lightGrey,
        paddingLeft: 5,
        paddingRight: 10
    },
    textContainer: {
        flex: 4,
        flexDirection: "column",
        paddingVertical: 10,
        paddingHorizontal: 15,
        alignItems: "flex-start"
    },
    imageContainer: {
        justifyContent: "center",
        alignItems: "center"
    },
    image: {
        height: 30,
        width: 30,
        borderRadius: 30 / 2
    }
});
