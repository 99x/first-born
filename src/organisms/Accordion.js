import React, { Component } from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";
import { AccordionElement } from "../molecules/cards/AccordionElement";

export class Accordion extends Component {
    render() {
        const { horizontal, renderItem, ...otherProps } = this.props;

        return (
            <FlatList
                keyExtractor={this.keyExtractor}
                renderItem={
                    renderItem
                        ? renderItem
                        : ({ item }) => {
                              return (
                                  <AccordionElement
                                      {...item}
                                      isList
                                      {...otherProps}
                                  />
                              );
                          }
                }
                {...otherProps}
            />
        );
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

Accordion.propTypes = {
    bodyColor: PropTypes.string,
    headerColor: PropTypes.string,
    expandedIcon: PropTypes.string,
    icon: PropTypes.string,
    ...FlatList.propTypes
};
