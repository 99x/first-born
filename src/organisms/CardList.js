import React, { Component } from "react";
import { FlatList, Dimensions, View, Platform } from "react-native";
import { Card } from "../molecules/cards/Card";

const { width } = Dimensions.get("window");

export class CardList extends Component {

    render() {
        const { backgroundColor, horizontal, margin, ...otherProps } = this.props;

        return (
            <FlatList
                style={{ marginTop: 5 }}
                keyExtractor={this.keyExtractor}
                renderItem={({ item }) => {
                    return (
                        <View style={{ marginRight: margin }}>
                            <Card image={item.image} title={item.title} description={item.description} backgroundColor={backgroundColor} style={{ width: width - 40 }} />
                        </View>
                    )
                }}
                horizontal
                {...otherProps} />
        )
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

CardList.propTypes = {
    ...FlatList.propTypes
}

CardList.defaultProps = {
    margin: Platform.OS === "ios" ? 15 : 10
}
