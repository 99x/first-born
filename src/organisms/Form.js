import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../atoms/Picker";

import { FormDatePicker } from "../molecules/formElements/FormDatePicker";
import { FormInput } from "../molecules/formElements/FormInput";
import { FormPicker } from "../molecules/formElements/FormPicker";
import { FormTextArea } from "../molecules/formElements/FormTextArea";

export class Form extends Component {
    render() {
        const { color, formElements, ...otherProps } = this.props;

        const inputColor = color ? color : "#486c86";

        return (
            <View style={styles.container}>
                {formElements.map((element, key) => {
                    return (
                        <View style={styles.formElementContainer} key={key}>
                            {this.renderElements(element, inputColor, otherProps)}
                        </View>
                    );
                })}
            </View>
        );
    }

    renderElements(element, color, styleProps) {
        element.color = color;
        const { type, pickerData, ...otherProps } = element;

        switch (type) {
            case "text":
                return <FormInput {...styleProps} {...otherProps} />;
            case "textarea":
                return <FormTextArea {...styleProps} {...otherProps} />;
            case "date":
                return <FormDatePicker {...styleProps} {...otherProps} />;
            case "picker":
                return (
                    <FormPicker {...styleProps} {...otherProps}>
                        {pickerData.map((dataElement, key) => (
                            <Picker.Item {...dataElement} key={key} />
                        ))}
                    </FormPicker>
                );
        }
    }
}

Form.propTypes = {
    color: PropTypes.string,
    activeStyle: PropTypes.object,
    errorStyle: PropTypes.object,
    rounded: PropTypes.bool,
    underline: PropTypes.bool,
    defaultStyle: PropTypes.bool,
    noStyle: PropTypes.bool,
    errorColor: PropTypes.string,
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: "100%"
    },
    formElementContainer: {
        marginTop: 10
    }
});
