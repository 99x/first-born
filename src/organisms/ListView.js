import React, { Component } from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";
import { ListItem } from "../molecules/listItem/ListItem";
import { ThinListItem } from "../molecules/listItem/ThinListItem";

export class ListView extends Component {
    render() {
        const {
            backgroundColor,
            horizontal,
            renderItem,
            secondary,
            thin,
            rounded,
            ...otherProps
        } = this.props;

        return (
            <FlatList
                style={{ width: "100%" }}
                keyExtractor={this.keyExtractor}
                renderItem={
                    renderItem
                        ? renderItem
                        : ({ item }) => {
                              if (thin) {
                                  return (
                                      <ThinListItem
                                          {...item}
                                          backgroundColor={backgroundColor}
                                          secondary={secondary}
                                          rounded={rounded}
                                      />
                                  );
                              }
                              return (
                                  <ListItem
                                      {...item}
                                      backgroundColor={backgroundColor}
                                      secondary={secondary}
                                      rounded={rounded}
                                  />
                              );
                          }
                }
                {...otherProps}
            />
        );
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

ListView.propTypes = {
    backgroundColor: PropTypes.string,
    thin: PropTypes.bool,
    rounded: PropTypes.bool,
    ...FlatList.propTypes
};

ListView.defaultProps = {
    thin: false,
    rounded: true
};
