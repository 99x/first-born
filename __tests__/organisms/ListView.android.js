import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { ListView } from "../../src/organisms/ListView";
import { ListItem } from "../../src/molecules/listItem/ListItem";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders a list of listItems with heading", () => {
    const listData = [
        { title: "Heading 1" },
        { title: "Heading 2" },
        { title: "Heading 3" }
    ];

    const tree = renderer.create(<ListView data={listData} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of listItems with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer.create(<ListView data={listData} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of listItems with heading, title and image", () => {
    const listData = [
        {
            title: "Heading 1",
            description: "Description 1",
            image: { source: require("../../firstBorn-logo.png") }
        },
        {
            title: "Heading 2",
            description: "Description 2",
            image: { source: require("../../firstBorn-logo.png") }
        },
        {
            title: "Heading 3",
            description: "Description 3",
            image: { source: require("../../firstBorn-logo.png") }
        }
    ];

    const tree = renderer.create(<ListView data={listData} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a colored list of listItems with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(<ListView data={listData} backgroundColor="aliceblue" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of thin listItems with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer.create(<ListView data={listData} thin />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of thin listItems with heading, title and image", () => {
    const listData = [
        {
            title: "Heading 1",
            description: "Description 1",
            image: { source: require("../../firstBorn-logo.png") }
        },
        {
            title: "Heading 2",
            description: "Description 2",
            image: { source: require("../../firstBorn-logo.png") }
        },
        {
            title: "Heading 3",
            description: "Description 3",
            image: { source: require("../../firstBorn-logo.png") }
        }
    ];

    const tree = renderer.create(<ListView data={listData} thin />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a colored list of thin listItems with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(<ListView data={listData} backgroundColor="aliceblue" thin />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of listItems with heading and title by a user defined renderItem method", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(
            <ListView
                data={listData}
                renderItem={({ item }) => <ListItem {...item} />}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
