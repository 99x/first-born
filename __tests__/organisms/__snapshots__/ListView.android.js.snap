// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a colored list of cards with heading and title 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Description 1",
        "title": "Heading 1",
      },
      Object {
        "description": "Description 2",
        "title": "Heading 2",
      },
      Object {
        "description": "Description 3",
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 5,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 2
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders a list of cards with heading 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "title": "Heading 1",
      },
      Object {
        "title": "Heading 2",
      },
      Object {
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 5,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders a list of cards with heading and title 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Description 1",
        "title": "Heading 1",
      },
      Object {
        "description": "Description 2",
        "title": "Heading 2",
      },
      Object {
        "description": "Description 3",
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 5,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 2
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderBottomColor": "rgba(33, 33, 33, 0.4)",
              "borderBottomWidth": 0.5,
              "borderTopColor": "rgba(33, 33, 33, 0.4)",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 3,
                "flexDirection": "column",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
